extension/background/serviceList.js

```diff
@@ -164,14 +164,29 @@ export class Service {
 }
 
 export async function detectServiceFromActiveTab(services) {
+  let serviceName = null;
   const tab = await browserUtil.activeTab();
   for (const name in services) {
     const service = services[name];
     if (tab.url.startsWith(service.baseUrl)) {
-      return name;
+      if (service.necessaryQuery === undefined) {
```

Comment:
OK, I think I understand the code here, but some comments or different names might be in order.

`necessaryQuery` is necessary _here_, but if I don't know what your intention is it's not so clear. I think it would be clearer if it was named using `baseUrl`, since it's an additional filter to what `baseUrl` is used for. Like `baseUrlQueryParameters`